File: shared/contracts/python/v1/backlink.py
from __future__ import annotations

from pydantic import BaseModel, Field


class BacklinkItem(BaseModel):
    source_note_id: str = Field(min_length=1)
    source_note_title: str = Field(min_length=1)
    matched_title: str = Field(min_length=1)
    snippet: str = Field(min_length=1)
    updated_at: str = Field(min_length=1)


class BacklinksResponse(BaseModel):
    note_id: str = Field(min_length=1)
    items: list[BacklinkItem]

File: shared/contracts/ts/v1/backlink.ts
export interface BacklinkItem {
  source_note_id: string;
  source_note_title: string;
  matched_title: string;
  snippet: string;
  updated_at: string;
}

export interface BacklinksResponse {
  note_id: string;
  items: BacklinkItem[];
}
